design.md

```diff
@@ -2,12 +2,56 @@
 
 ## Who is the target for this design, e.g., are you assuming any knowledge on the part of the language users?
 
+We assume some familiarity with programming, because we use a code block-like
+syntax of rules (as a grouped-by-state structure).
+
 ## Why did you choose this design, i.e., why did you think it would be a good idea for users to express the maze-searching computation using this syntax?
 
+Picobot has a lot of redundancy in that the states need to mentioned on a per
+rule basis in the original syntax. We both would group these rules together
+anyway when writing in the old syntax, so it seemed helpful to enforce this.
```

Comment:
I agree. Even the original Picobot sample solution groups states together separated by whitespace, so why not enforce it to decrease redundancy.